app/scenes/Document/components/Comments.tsx

```diff
@@ -50,19 +56,62 @@ function Comments() {
     : { type: CommentSortType.MostRecent };
 
   const viewingResolved = params.get("resolved") === "";
-  const resolvedThreads = document
+  const threads = !document
+    ? []
+    : viewingResolved
     ? comments.resolvedThreadsInDocument(document.id, sortOption)
-    : [];
+    : comments.unresolvedThreadsInDocument(document.id, sortOption);
+  const hasComments = threads.length > 0;
+
+  const scrollToBottom = () => {
+    if (scrollableRef.current) {
+      scrollableRef.current.scrollTo({
+        top: scrollableRef.current.scrollHeight,
+      });
+    }
+  };
+
+  const handleScroll = () => {
+    const BUFFER_PX = 50;
+
+    if (scrollableRef.current) {
+      const sh = scrollableRef.current.scrollHeight;
+      const st = scrollableRef.current.scrollTop;
+      const ch = scrollableRef.current.clientHeight;
+      isAtBottom.current = Math.abs(sh - (st + ch)) <= BUFFER_PX;
+
+      if (isAtBottom.current) {
+        setShowJumpToRecentBtn(false);
+      }
+    }
+  };
+
+  React.useEffect(() => {
+    // Handles: 1. on refresh 2. when switching sort setting
+    if (readyToDisplay && sortOption.type === CommentSortType.MostRecent) {
+      scrollToBottom();
+    }
+  }, [sortOption.type, readyToDisplay]);
+
+  React.useEffect(() => {
+    setShowJumpToRecentBtn(false);
+    if (sortOption.type === CommentSortType.MostRecent) {
+      const commentsAdded = threads.length > prevThreadCount.current;
+      if (commentsAdded) {
+        if (isAtBottom.current) {
+          scrollToBottom(); // Remain pinned to bottom on new comments
+        } else {
+          setShowJumpToRecentBtn(true);
+        }
+      }
+    }
+    prevThreadCount.current = threads.length;
+  }, [threads.length]);
 
-  if (!document || !isEditorInitialized) {
```

Comment:
Why did you remove the `document` guard in favor of lots of non-null assertion? Seems objectively worse 🤔

Reply:
The null guard was DRY-ed for re-use as an effect dependency, but Typescript for some reason can't figure out that `document` will never be null, resulting in the need to suppress the check everywhere.

Agree that it's more messy, so un-DRY-ed